openshift/openshift/src/main/java/org/arquillian/cube/openshift/impl/resources/OpenShiftResourceFactory.java

```diff
@@ -126,6 +131,57 @@ private static void createOpenShiftResources(String resourcesKey, OpenShiftAdapt
         }
     }
 
+    private static void createDynamicOpenShiftImageStreamResources(String resourcesKey, OpenShiftAdapter adapter, Class<?> testClass,
+                                                                   StringResolver resolver, List<OpenShiftDynamicImageStreamResource> openShiftDynamicImageStreamResources) throws IOException {
+        for (OpenShiftDynamicImageStreamResource imageStreamResource : openShiftDynamicImageStreamResources) {
+            String name = resolver.resolve(imageStreamResource.name());
+            String image = resolver.resolve(imageStreamResource.image());
+            String version = resolver.resolve(imageStreamResource.version());
+            boolean insecure = Boolean.valueOf(resolver.resolve(imageStreamResource.insecure()));
+
+            JSONObject imageStream = new JSONObject();
+            JSONObject metadata = new JSONObject();
+            JSONObject annotations = new JSONObject();
+
+            metadata.put("name", name);
+            annotations.put("openshift.io/image.insecureRepository", insecure);
+            metadata.put("annotations", annotations);
+
+            // Definition of the image
+            JSONObject from = new JSONObject();
+            from.put("kind", "DockerImage");
+            from.put("name", image);
+
+            JSONObject importPolicy = new JSONObject();
+            importPolicy.put("insecure", insecure);
+
+            JSONObject tag = new JSONObject();
+            tag.put("name", version);
+            tag.put("from", from);
+            tag.put("importPolicy", importPolicy);
+
+            JSONObject tagAnnotations = new JSONObject();
+            tagAnnotations.put("version", version);
+            tag.put("annotations", tagAnnotations);
+
+            JSONArray tags = new JSONArray();
+            tags.add(tag);
+
+            // Add image definition to image stream
+            JSONObject spec = new JSONObject();
+            spec.put("tags", tags);
+
+            imageStream.put("kind", "ImageStream");
+            imageStream.put("apiVersion", "v1");
+            imageStream.put("metadata", metadata);
+            imageStream.put("spec", spec);
+
+            log.info(String.format("Creating new OpenShift image stream resource: '%s'", imageStream.toJSONString()));
```

Comment:
Can you add the Json creation inside a method? Maybe called like `createImageStreamRequest` or sth similar.